opentracing-api/src/main/java/io/opentracing/propagation/BinaryAdapters.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.propagation;
+
+import java.nio.ByteBuffer;
+
+final class BinaryAdapters {
+
+    private BinaryAdapters() {}
+
+    public static class BinaryExtractAdapter implements Binary {
```

Comment:
does it have to be public?

Reply:
In the current way, yes. But we can hide them through a single method (as we do now for `Adapters` ;) )